examples/datacalc.py

```diff
@@ -67,10 +67,12 @@ def __init__(self, name: str, tables: Dict[str, str], parent: Optional[QObject]
         self.dbs = {"": ""}
         self.dbNames = {"A": "", "B": ""}
         self.viewerFrame = ViewerFrame()
+        for dbBox in self.viewerFrame.dbBoxes.values():
+            dbBox.addItem("")
         # connect signals to slots
         self.received.connect(self.updateDB)
-        for dbBox in self.viewerFrame.dbBoxes.values():
-            dbBox.currentIndexChanged.connect(self.setDB)
+        self.viewerFrame.dbBoxes["A"].currentIndexChanged.connect(lambda: self.setDB("A"))
+        self.viewerFrame.dbBoxes["B"].currentIndexChanged.connect(lambda: self.setDB("B"))
```

Comment:
I set a slot of each combobox individually, because the following code doesn't work correctly.
```python
for dbName, dbBox in self.viewerFrame.dbBoxes.items():
    dbBox.currentIndexChanged.connect(lambda: self.setDB(dbName))
```

In the code, two slots are set as `lambda: self.setDB("B")`, not separately.
I think it may be a problem in terms of reference and I tried to fix it, but I couldn't make it.

Reply:
In that case, you can use [`functools.partial()`](https://docs.python.org/3/library/functools.html#functools.partial):
```python
functools.partial(self.setDB, dbName) # instead of lambda
```
If you use the `lambda` expression like that, it actually defines a function, and hence refers to `dbName` _when the function is called_.
Therefore the both dbBox A and B will be connected to technically the same slots, since the both slots refer to the same variable `dbName`.

However, if you use `functools.partial()`, the _value_ of `dbName` is passed when `partial()` function is called (i.e., just before the slot is connected).

Reply:
Wow.. I've been trying so hard to solve this. Thanks for the advice😂